keepassxc-browser/content/totp-field.js

```diff
@@ -48,6 +48,12 @@ kpxcTOTPIcons.isAcceptedTOTPField = function(field) {
     return false;
 };
 
+// Double the allowed maxLength if an exception is found (e.g. Amazon's 2FA)
+kpxcTOTPIcons.isLengthValid = function(field) {
+    const maxLength = kpxcSites.exceptionFound(PREDEFINED_TOTP) ? MAX_INPUT_LENGTH * 2 : MAX_INPUT_LENGTH;
+    return (field.maxLength > 0 && (field.maxLength < MIN_INPUT_LENGTH || field.maxLength > maxLength));
+};
```

Comment:
This is a very specific action (double the length) for a generic response (there is an exception). I would much prefer to not re-use the exception function which is already rather ungainly, but instead have a function such as "expectedTotpLength(site_url)" that would return the default max length in all cases except for Amazon (for now).

Reply:
Good idea. I'll make the necessary changes.